ui/src/CheckSources.js

```diff
@@ -35,22 +40,22 @@ const CheckSources = React.createClass({
   getInitialState() {
     return {
       isFetching: true,
-      sources: [],
     };
   },
 
   componentDidMount() {
     getSources().then(({data: {sources}}) => {
-      this.setState({sources, isFetching: false});
+      this.props.loadSourcesAction(sources);
```

Comment:
I personally like the idea of putting the AJAX call into the action itself. Then the action would look like this https://github.com/influxdata/chronograf/blob/master/ui/src/kapacitor/actions/view/index.js#L46

Let me know if that makes no sense